cockatrice/src/settingscache.cpp

```diff
@@ -612,4 +613,12 @@ void SettingsCache::setNotifyAboutUpdate(int _notifyaboutupdate)
 {
     notifyAboutUpdates = _notifyaboutupdate;
     settings->setValue("personal/updatenotification", notifyAboutUpdates);
+}
+
+QString SettingsCache::getSrvClientID(const QString _hostname)
+{
+    QString srvClientID = getClientID();
+    srvClientID += _hostname;
+    QString uniqueServerClientID = QCryptographicHash::hash(srvClientID.toUtf8(), QCryptographicHash::Sha1).toHex().right(15);
```

Comment:
So one of the original concerning factors of the client-id functionality was that since the id was being generated off hardware devices on a persons machine that if we sent the hash to some were that some how some one may be able to reverse engineer the information and obtain the original senders hardware id.  So to mitigate the concern as much as possible we only took a portion of the generated hash and passed in hopes that the potential for some how reversing the hashed information back to some one's physical address list off there computer wouldnt be possible.

Also one thing to note, we have a varchar(15) in the db table so anything longer than that would just get truncated anyway.